comp/core/workloadmeta/collectors/internal/containerd/image.go

```diff
@@ -315,15 +318,21 @@ func (c *collector) notifyEventForImage(ctx context.Context, namespace string, i
 	// if the `imgMeta` object does not contain all the metadata when it is sent.
 	// We add them here to make sure they are present.
 	wlmImage.SBOM = util.UpdateSBOMRepoMetadata(wlmImage.SBOM, wlmImage.RepoTags, wlmImage.RepoDigests)
+	return &wlmImage, nil
+}
 
+func (c *collector) notifyEventForImage(ctx context.Context, namespace string, img containerd.Image, sbom *workloadmeta.SBOM) error {
+	wlmImage, err := c.createOrUpdateImageMetadata(ctx, namespace, img, sbom)
+	if err != nil {
+		return err
+	}
 	c.store.Notify([]workloadmeta.CollectorEvent{
 		{
 			Type:   workloadmeta.EventTypeSet,
 			Source: workloadmeta.SourceRuntime,
-			Entity: &wlmImage,
+			Entity: wlmImage,
 		},
 	})
-
 	return nil
 }
```

Comment:
do we still need this function?

Reply:
Notify has been moved out from createOrUpdateImageMetadata function.